src/rgw/rgw_lc.cc

```diff
@@ -0,0 +1,546 @@
+  result = split(shard_id, ':');
+  string bucket_tenant = result[0];
+  string bucket_name = result[1];
+  string bucket_id = result[2];
+  int ret = store->get_bucket_info(obj_ctx, bucket_tenant, bucket_name, bucket_info, NULL, &bucket_attrs);
+  if (ret < 0) {
+    ldout(cct, 0) << "LC:get_bucket_info failed" << bucket_name <<dendl;
+    return ret;
+  }
+
+  ret = bucket_info.bucket.bucket_id.compare(bucket_id) ;
+  if (ret !=0) {
+    ldout(cct, 0) << "LC:old bucket id find, should be delete" << bucket_name <<dendl;
+    return -ENOENT;
+  }
+  
+  RGWRados::Bucket target(store, bucket_info);
+  RGWRados::Bucket::List list_op(&target);
+
+  list_op.params.prefix = prefix;
+  list_op.params.delim = delimiter;
+  list_op.params.marker = marker;
+  list_op.params.end_marker = end_marker;
+  list_op.params.list_versions = false;
+   
+  map<string, bufferlist>::iterator aiter = bucket_attrs.find(RGW_ATTR_LC);
+  if (aiter == bucket_attrs.end())
+    return 0;
+  
+  bufferlist::iterator iter(&aiter->second);
+  try {
+      config.decode(iter);
+    } catch (const buffer::error& e) {
+      ldout(cct, 0) << __func__ <<  "decode life cycle config failed" << dendl;
+      return -1;
+    }
+
+  map<string, int>& prefix_map = config.get_prefix_map();
+  for(map<string, int>::iterator prefix_iter = prefix_map.begin(); prefix_iter != prefix_map.end();  prefix_iter++) {
+    if (prefix_iter->first.empty()) {
+      default_config = true;
+      default_days = prefix_iter->second;
+      continue;
+    }
+  }
+
+  if (default_config) { 
+    do {
+      
+      objs.clear();
+      list_op.params.marker = list_op.get_next_marker();
+      ret = list_op.list_objects(1000, &objs, &common_prefixes, &is_truncated);    
+      if (ret < 0) {
+        if (ret == -ENOENT)
+          return 0;
+        ldout(cct, 0) << "ERROR: store->list_objects():" <<dendl;
+        return ret;
+      }
+
+      vector<RGWObjEnt>::iterator obj_iter;
+      int pos = 0;
+      utime_t now = ceph_clock_now(cct);
+      for (obj_iter = objs.begin(); obj_iter != objs.end(); obj_iter++) {
+        bool prefix_match = false;
+        int match_days = 0;        
+        map<string, int>& prefix_map = config.get_prefix_map();
+        
+        for(map<string, int>::iterator prefix_iter = prefix_map.begin(); prefix_iter != prefix_map.end();  prefix_iter++) {
+          if (prefix_iter->first.empty()) {
+            continue;
+          }
+          pos = (*obj_iter).key.name.find(prefix_iter->first, 0);
+          if (pos != 0) {
+            continue;
+          }
+          prefix_match = true;
+          match_days = prefix_iter->second;
+          break;
+        }
+        int days = 0;      
+        if (prefix_match) {
+          days = match_days;
+        } else if (default_config) {
+          days = default_days;
+        } else {
+          continue;
+        }        
+        if (now - ceph::real_clock::to_time_t((*obj_iter).mtime) >= days*24*60*60) {
```

Comment:
@dang 
what is the result of (time_point_A - time_point_B), and how to compare it with seconds?